### ISSUE TITLE
Add new feature to delete list of attributes by ID

### ISSUE THREAD

**Issue by chrisinmtown** [MISP/PyMISP#689]
Request a PyMISP method that accepts a list of attributes to be deleted. In version 2.4.133 api.py exposes method `delete_attribute(id)` which works fine. But is not sustainable when the task is to delete 100,000 or other horribly large number of attributes from an event. In my tests, calling that method in a tight loop only deletes a couple hundred attributes per minute.  We need this feature to clean the mess of soft-deleted attributes left by the delta-merge algorithm on fixed-event feeds.

I'm hoping this is trivial bcos apparently the MISP server supports this already according to this comment thread:

https://github.com/MISP/MISP/issues/6010#issuecomment-660891990

What's the right way - extend the delete_attribute method to accept a list or add a new method?  If you will please reply with the team's preference then I will make a proposal for a change to api.py.  Thanks for listening.

**Comment by chrisinmtown** [comment]
I have to correct what I wrote above.  The rate for deleting attributes from an event seems to vary with the number of attributes on the event.  A PyMISP script can delete multiple attributes per second from an event with a small (fewer than 1,000) attributes.  But if the event has 100K attributes, a delete operation can take 5..10 seconds. I don't know what is responsible for the large execution-time cost, if it's locking, queries or whatever.  I hope the CIRCL team can advise if a single delete-id-list function would be faster than successive REST calls.

**Comment by Rafiot** [comment]
Yeah, this method is meant for the UI, not for the API, we need to implement something specific for the API. My guess regarding the time is that as it deletes attributes sequentially, and each deleted attribute triggers an update in the correlation table.

**Comment by chrisinmtown** [comment]
Well maybe if the team selects this MISP extension for development, the PyMISP feature will be easy: https://github.com/MISP/MISP/issues/6934

**Comment by github-germ** [comment]
Related... https://github.com/MISP/MISP/issues/6934#issuecomment-1847508251

**Comment by github-germ** [comment]
@iglocska This is an important use case for us. Let us know if we can provide any addition info to help move this feature request forward. Thanks.

**Comment by github-germ** [comment]
@iglocska Hi... revisiting this with 2.4.191 -- any chance that we can see this feature sometime soon?

**Comment by github-germ** [comment]
Revisiting this 3+ year old issue now that 2.4.196 has been released. Are there any plans to implement, and if so, any targeted timeframe? Thanks!

**Comment by eviltik** [comment]
3 years. This is a joke.
